orm/helpers.go

```diff
@@ -0,0 +1,70 @@
+package orm
+
+import (
+	"reflect"
+
+	"github.com/iov-one/weave/errors"
+)
+
+type limitedIterator struct {
+	// n is count of remaining elements
+	n int
+	// iterator is the underlying iterator
+	iter SerialModelIterator
+}
+
+var _ SerialModelIterator = (*limitedIterator)(nil)
+
+func (l *limitedIterator) LoadNext(dest SerialModel) error {
+	if l.n > 0 {
+		l.n--
+		return l.iter.LoadNext(dest)
```

Comment:
I think you should not decrement the counter if the iterator returned an error. The current approach can produce an unexpected result:
```
it := Limit(bucket, 2)
val1, err := next(it) // ok
_, err := next(it) // temporary error
val2, err := next(it) // ErrIteratorDone, although I expect a second value